### PR TITLE
cluster-display: cache page data

### PR DESCRIPTION
**Pull request by hongkailiu**

### CODE REVIEW

File: cmd/cluster-display/main.go [hunk 1]
```diff
@@ -77,7 +78,8 @@ func addSchemes() error {
 	return nil
 }
 
-func getClusterPoolPage(ctx context.Context, hiveClient ctrlruntimeclient.Client) (*Page, error) {
+func getClusterPool(ctx context.Context, hiveClient ctrlruntimeclient.Client) ([]map[string]string, error) {
+	logrus.Info("Calling getClusterPool ...")
```
Comment (smg247) [inline cmd/cluster-display/main.go:82]: If you meant to leave this log in, I think it should be `debug` level

File: cmd/cluster-display/main.go [hunk 2]
```diff
@@ -161,13 +162,62 @@ func (g *clusterInfoGetter) GetClusterDetails(ctx context.Context, cluster strin
...
+	ClusterPoolData              []map[string]string
+	ClusterPoolDataLastUpdatedAt time.Time
+}
+
+func (c *memoryCache) GetClusterPoolPage(ctx context.Context, client ctrlruntimeclient.Client) (*Page, error) {
+	c.ClusterPoolDataMutex.Lock()
+	defer c.ClusterPoolDataMutex.Unlock()
+	if c.ClusterPoolData == nil || time.Now().After(c.ClusterPoolDataLastUpdatedAt.Add(time.Hour)) {
+		data, err := getClusterPool(ctx, client)
+		if err != nil {
```
Comment (smg247) [inline cmd/cluster-display/main.go:180]: We return an error here, but some of the value I was thinking we could achieve by adding the cache is to continue to return the previous results in this state. That way we could still display the last available information about the cluster. We could maybe display a warning that there is a problem, and one or more cluster may be unreachable or outdated along with this info.

File: cmd/cluster-display/main.go (hunk 1 above)
Comment (hongkailiu) [reply cmd/cluster-display/main.go]: Done

File: cmd/cluster-display/main.go (hunk 2 above)
Comment (hongkailiu) [reply cmd/cluster-display/main.go:182]: > We could maybe display a warning that there is a problem

Added a log in the backend instead.

**Pull request merged**